### PR TITLE
barman: fix build on darwin

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

ZHF #199919.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/barman/default.nix [hunk 1]
```diff
@@ -34,6 +34,8 @@ python3Packages.buildPythonApplication rec {
   ];
 
   disabledTests = [
+    # FsOperationFailed
+    "test_get_file_mode"
```
Comment (Atemu) [inline pkgs/tools/misc/barman/default.nix:38]: Only on Darwin? If so, only disable the check on Darwin.

**Review by Atemu — COMMENTED** [review]
Result of `nixpkgs-review pr 199276` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>barman</li>
  </ul>
</details>

Ran `barman --help`.

File: pkgs/tools/misc/barman/default.nix (hunk 1 above)
Comment (wegank) [reply pkgs/tools/misc/barman/default.nix:39]: Done.

File: pkgs/tools/misc/barman/default.nix [hunk 2]
```diff
@@ -34,6 +35,9 @@ python3Packages.buildPythonApplication rec {
...
+  ] ++ lib.optionals stdenv.isDarwin [
```
Comment (Atemu) [inline pkgs/tools/misc/barman/default.nix:40]: This is the wrong way around.

**Review by Atemu — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/barman/default.nix (hunk 2 above)
Comment (wegank) [reply pkgs/tools/misc/barman/default.nix:40]: Oh my fault!

Comment (wegank) [reply pkgs/tools/misc/barman/default.nix]: Done.

File: pkgs/tools/misc/barman/default.nix [hunk 3]
```diff
@@ -1,5 +1,6 @@
 { fetchFromGitHub
 , lib
+, stdenv
```
Comment (Atemu) [inline pkgs/tools/misc/barman/default.nix:3]:
```suggestion
```

File: pkgs/tools/misc/barman/default.nix [hunk 4]
```diff
@@ -36,6 +37,9 @@ python3Packages.buildPythonApplication rec {
   disabledTests = [
     # Assertion error
     "test_help_output"
+  ] ++ lib.optionals stdenv.isDarwin [
```
Comment (Atemu) [inline pkgs/tools/misc/barman/default.nix:40]:
```suggestion
  ] ++ lib.optionals python3Packages.stdenv.isDarwin [
```

**Review by Atemu — APPROVED** [review]

File: pkgs/tools/misc/barman/default.nix (hunk 4 above)
Comment (wegank) [reply pkgs/tools/misc/barman/default.nix:40]: Having not found `python3Packages.stdenv` in nixpkgs, I doubt it will work...

Comment (Atemu) [reply pkgs/tools/misc/barman/default.nix:40]: `python3Packages.stdenv` evals to a stdenv for me.

**Pull request merged**